pkg/tagger/collectors/docker_extract_test.go

```diff
@@ -0,0 +1,130 @@
+)
+
+func TestDockerRecordsFromInspect(t *testing.T) {
+	testCases := []struct {
+		co                   *types.ContainerJSON
+		toRecordEnvAsTags    map[string]string
+		toRecordLabelsAsTags map[string]string
+		expectedLow          []string
+		expectedHigh         []string
+	}{
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelValue"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{},
+			toRecordLabelsAsTags: map[string]string{},
+			expectedLow:          []string{},
+			expectedHigh:         []string{},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelKey"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "becomeK"},
+			toRecordLabelsAsTags: map[string]string{"labelKey": "labelValue"},
+			expectedLow:          []string{"becomeK:v"},
+			expectedHigh:         []string{},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v"},
+					Labels: map[string]string{"labelKey": "labelKey"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "+becomeK"},
+			toRecordLabelsAsTags: map[string]string{"labelKey": "labelValue"},
+			expectedLow:          []string{},
+			expectedHigh:         []string{"becomeK:v"},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v", "l=t"},
+					Labels: map[string]string{"labelKey": "labelValue"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "+becomeK", "l": "expectedLow"},
+			toRecordLabelsAsTags: map[string]string{"labelkey": "labelKey"},
+			expectedLow:          []string{"expectedLow:t", "labelKey:labelValue"},
+			expectedHigh:         []string{"becomeK:v"},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env:    []string{"k=v", "l=t"},
+					Labels: map[string]string{"labelKey": "labelValue"},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{"k": "+becomeK", "l": "expectedLow"},
+			toRecordLabelsAsTags: map[string]string{"labelkey": "+labelKey"},
+			expectedLow:          []string{"expectedLow:t"},
+			expectedHigh:         []string{"becomeK:v", "labelKey:labelValue"},
+		},
+		{
+			co: &types.ContainerJSON{
+				Config: &container.Config{
+					Env: []string{
+						"MARATHON_APP_ID=1",
+						"CHRONOS_JOB_NAME=2",
+						"CHRONOS_JOB_OWNER=3",
+						"MESOS_TASK_ID=4",
+					},
+					Labels: map[string]string{},
+				},
+			},
+			toRecordEnvAsTags:    map[string]string{},
+			toRecordLabelsAsTags: map[string]string{},
+			expectedLow: []string{
+				"marathon_app:1",
+				"chronos_job:2",
+				"chronos_job_owner:3",
+			},
+			expectedHigh: []string{"mesos_task:4"},
+		},
+	}
+
+	dc := &DockerCollector{}
+	for i, test := range testCases {
+		dc.envAsTags = test.toRecordEnvAsTags
```

Comment:
please use the sub-test patern for easier debugging: https://github.com/DataDog/datadog-agent/blob/master/pkg/util/docker/global_test.go#L49

I'll grep and migrate all the test using this old pattern in a +1 week